ecl/hqlcpp/hqltcppc.cpp

```diff
@@ -820,17 +820,24 @@ void CMemberInfo::checkAssignOk(HqlCppTranslator & translator, BuildCtx & ctx, I
                         combined = true;
                     }
                 }
-
-                if (!combined)
+            }
+            else
+            {
+                if (match)
                 {
-                    IHqlStmt * stmt = ctx.addExpr(call);
-                    ctx.associateOwn(* new HqlStmtExprAssociation(marker, stmt));
-                    combined = true;
+                    IHqlExpression * prevCall = match->stmt->queryExpr(0);
+                    //If the builder and the size are the same there is no need to call again (ignore the field name)
+                    if ((call->queryChild(1) == prevCall->queryChild(1)) && (call->queryChild(0) == prevCall->queryChild(0)))
+                        combined = true;
                 }
             }
             
             if (!combined)
-                ctx.addExpr(call);
+            {
+                IHqlStmt * stmt = ctx.addExpr(call);
+                ctx.associateOwn(* new HqlStmtExprAssociation(marker, stmt));
+                combined = true;
```

Comment:
@ghalliday minor: combined shouldn't really be set to true here.

Reply:
You are correct - I will delete it.